### DISCUSSION TITLE
How to calculate one zone in automation

### DISCUSSION THREAD

**Discussion by JiriPrchal** [jeroenterheerdt/HAsmartirrigation#460] [Q&A] [answered]
Hi all,
I think this simple automation should work:
` - alias: smart irrigation greenhouse
   trigger:
    - platform: numeric_state
      entity_id: sun.sun
      attribute: elevation
      above: 0
   condition:
    - condition: template
      value_template: '{{ (as_timestamp (now ()) | timestamp_custom ("%j") | int % 3) == 0 }}'
   action:
    - service: smart_irrigation.calculate_zone
      entity_id: zone.smart_irrigation_greenhouse
    - service: switch.turn_on
      entity_id: switch.greenhouse_irrigation
    - delay: '{{ states (sensor.smart_irrigation_greenhouse) }}'
    - service: switch.turn_off
      entity_id: switch.greenhouse_irrigation
    - service: smart_irrigation.reset_bucket
      entity_id: zone.smart_irrigation_greenhouse
`
But there is an error:
`ERROR (MainThread) [homeassistant.helpers.template] Template variable error: 'sensor' is undefined when rendering '{{ states (sensor.smart_irrigation_greenhouse) }}'
ERROR (MainThread) [homeassistant.components.automation.smart_irrigation_greenhouse] smart irrigation greenhouse: Error rendering smart irrigation greenhouse delay template: UndefinedError: 'sensor' is undefined
`
and time, bucket are still 0.

**Comment by jeroenterheerdt** [comment]
Hard to parse but one issue I see if that you are referring to a zone entity that does not exist.

**Reply by JiriPrchal** [reply]
Yes, you are right. Everything with smart irrigation must be done to sensor? Weird, sensor is input usually.
Now its calculating, but not switched off and reset bucket:
```
- alias: smart irrigation greenhouse
   mode: single
   trigger:
    - platform: numeric_state
      entity_id: sun.sun
      attribute: elevation
      above: 0
   condition:
    - condition: template
      value_template: '{{ (as_timestamp (now ()) | timestamp_custom ("%j") | int % 3) == 0 }}'
   action:
    - service: smart_irrigation.calculate_zone
      entity_id: sensor.smart_irrigation_greenhouse
    - if:
       - condition: numeric_state
         entity_id: sensor.smart_irrigation_greenhouse
         above: 0
      then:
       - service: switch.turn_on
         entity_id: switch.greenhouse_irrigation
       - delay: '{{ states ("sensor.smart_irrigation_greenhouse") }}'
       - service: switch.turn_off
         entity_id: switch.greenhouse_irrigation
    - service: smart_irrigation.reset_bucket
      entity_id: senzor.smart_irrigation_greenhouse
```
Nothing in log :(

**Reply by JiriPrchal** [reply, marked as answer]
Am I a peace of fucking idiot, there is a typo "senzor". Rather to copy then type next time ;)
So now it's working:
```
 - alias: smart irrigation greenhouse
   mode: single
   trigger:
    - platform: numeric_state
      entity_id: sun.sun
      attribute: elevation
      above: 0
   condition:
    - condition: template
      value_template: '{{ (as_timestamp (now ()) | timestamp_custom ("%j") | int % 3) == 0 }}'
   action:
    - service: smart_irrigation.calculate_zone
      entity_id: sensor.smart_irrigation_greenhouse
    - if:
       - condition: numeric_state
         entity_id: sensor.smart_irrigation_greenhouse
         above: 0
      then:
       - service: switch.turn_on
         entity_id: switch.greenhouse_irrigation
       - delay: '{{ states ("sensor.smart_irrigation_greenhouse") }}'
       - service: switch.turn_off
         entity_id: switch.greenhouse_irrigation
    - service: smart_irrigation.reset_bucket
      entity_id: sensor.smart_irrigation_greenhouse
```
